blog/content/post/2021-12-12-optiplex-txt.md

```diff
@@ -0,0 +1,222 @@
+system bootloader. Now it is the bootloader's responsibility to perform the
+measured launch of the operating system by calling the SINIT ACM:
+
+![Breakout of measured launch details](/img/txt_launch2.jpg)
+
+> Source: *[A Practical Guide to TPM 2.0](https://link.springer.com/book/10.1007/978-1-4302-6584-9)*
+
+Currently there are two methods of executing SINIT ACM to perform measured
+launch:
+
+- [Trusted Boot (tboot)](https://sourceforge.net/projects/tboot/) - Intel's
+  reference implementation of Measured Launch Environment for Intel TXT
+- [GRUB2 with TrenchBoot support](https://trenchboot.org/) - GRUB2 is the most
+  popular open source bootloader. It can integrate TrenchBoot to perform
+  measured launch with Intel TXT or AMD Secure Startup.
+
+The above diagram represents the tboot case. Tboot is responsible here for
+checking all the prerequisites necessary to launch SINIT ACM and whether the
+BIOS did properly set up the Intel TXT (otherwise the measured launch will fail
+and we may anticipate something is wrong with the machine/firmware in such
+case). If all checks pass tboot loads an SINIT ACM, specifies the memory ranges
+containing the operating system software to be measured and executed the SINIT
+ACM using GETSEC\[SENTER\] instruction. Again the ACM signature is validated
+and ACM is being executed in an AC RAM (special secured RAM for ACM execution)
+to ensure nothing will tamper with the execution or measurement process. The
+ACM calculate the hashes of the operating system software and sends the mot the
+TPM Platform Configuration Registers (PCRs) which hold the hashes of the
+measured software. But they do not hold the hash itself, they combine the
+current register value and the hash of the software into a new hash which is
+saved into the register as a new value. This prevents easy faking of the PCRs
+(in order to achieve the same register value, all the measurements sent to the
+TPM must be done in the same order). BIOS does the same with BIOS code/modules,
+but how it is different in case of TXT?
+
+1. There are 24 PCRs inside TPM in total. BIOS can only extend PCRs 0 to 16 and
+   PCR 23. PCRs 17 to 22 can be only enabled by hardware, no software code is
+   able to initialize them, e.g. the microcode when launching the Intel TXT.
+   Before the PCR 17 to 22 are initialized their values are set to FFs (-1).
+   This is achieved by Intel TXT opening the TPM's locality 4, the only
+   locality able to reset the PCRs 17-22. Thsi locality is inaccessible by
+   software.
+2. SINIT ACM is being measured into PCR 17 as soon as TPM enters locality 4 via
+   GETSEC\[SENTER\] instruction execution. There is no possibility to alter
+   this.
+3. PCRs 0-16 can be only reset by system/TPM reset which will zero out these
+   PCRs and set the DRTM PCRs 17-22 to FFs (-1). PCRs 17-22 can be only reset
+   by GETSEC\[SENTER\] instruction which forces the SINIT ACM to be loaded and
+   perform a new measured launch.
```

Comment:
Overall it has to go to grammarly - I will get back to the review after you will pass it through checker.

Reply:
Grammarly done